tests/internal/test_tracer_flare.py

```diff
@@ -168,3 +172,114 @@ def test_no_app_logs(self):
+            }
+            self.tracer_flare_sub._get_data_from_connector_and_exec()
+
+    def test_process_flare_request_success(self):
+        """
+        Ensure a full successful tracer flare process
+        """
+        assert self.tracer_flare_sub.stale_tracer_flare_num_mins == 20
+        # Generate an AGENT_CONFIG product to trigger a request
+        with mock.patch("ddtrace.internal.flare.flare.Flare.prepare") as mock_flare_prep:
+            self.generate_agent_config()
+            mock_flare_prep.assert_called_once()
+
+        assert self.tracer_flare_sub.current_request_start is not None
+
+        # Generate an AGENT_TASK product to complete the request
+        with mock.patch("ddtrace.internal.flare.flare.Flare.send") as mock_flare_send:
+            self.generate_agent_task()
+            mock_flare_send.assert_called_once()
+
+        # Timestamp cleared after request completed
+        assert self.tracer_flare_sub.current_request_start is None
+
+    def test_detect_stale_flare(self):
+        """
+        Ensure we clean up and revert configurations if a tracer
+        flare job has gone stale
+        """
+        # Set this to 0 so all requests are stale
+        self.tracer_flare_sub.stale_tracer_flare_num_mins = 0
+
+        # Generate an AGENT_CONFIG product to trigger a request
+        with mock.patch("ddtrace.internal.flare.flare.Flare.prepare") as mock_flare_prep:
+            self.generate_agent_config()
+            mock_flare_prep.assert_called_once()
+
+        assert self.tracer_flare_sub.current_request_start is not None
+
+        # Setting this to 0 minutes so all jobs are considered stale
+        assert self.tracer_flare_sub.has_stale_flare()
+
+        self.generate_agent_config()
+
+        # Everything is cleaned up, reverted, and no current tracer flare
+        # timestamp
+        assert self.tracer_flare_sub.current_request_start is None
+
+    def test_no_overlapping_requests(self):
+        """
+        If a new tracer flare request is generated while processing
+        a pre-existing request, we will continue processing the current
+        one while disregarding the new request(s)
+        """
+        # Generate initial AGENT_CONFIG product to trigger a request
+        with mock.patch("ddtrace.internal.flare.flare.Flare.prepare") as mock_flare_prep:
+            self.generate_agent_config()
+            mock_flare_prep.assert_called_once()
+
+        original_request_start = self.tracer_flare_sub.current_request_start
+        assert original_request_start is not None
+
+        # Generate another AGENT_CONFIG product to trigger a request
+        # This should not be processed
+        with mock.patch("ddtrace.internal.flare.flare.Flare.prepare") as mock_flare_prep:
+            self.generate_agent_config()
+            mock_flare_prep.assert_not_called()
+
+        # Nothing should have changed, and we should still be processing
+        # only the original request
```

Comment:
Putting comments like these in the `assert` text (`assert foo == bar, "foo should be bar"`) adds a layer of usability to the tests.